### PR TITLE
Test api categories

### PR DESCRIPTION
**Pull request by at-dungle**
## Description
Test api categories
## Reference Requirements
- Task ID: TI-1

## Reference Isuees
#{issue_id}

## Migrations
YES | NO

## Impacted Areas in Application
List general components of the application that this PR will affect:

-

## SQL query and `explain`
List of all queries and attach the result of each explaining query (in text, screenshot...)

- Refer to [Using Explain with MySQL](https://viblo.asia/p/su-dung-explain-de-toi-uu-cau-lenh-mysql-BYjv44gmvxpV)

### CODE REVIEW

**Comment by at-sonle** [comment]
LGTM

**Comment by at-thinhnguyen** [comment]
Thêm case:
- Ko có data
- Search nếu có
- Phân trang nếu có

File: tests/Feature/Api/listCategoriesTest.php [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+                        'code'
+            ],
+            'data' => [
+                        'current_page',
+                        'data' => [
+                            [
+                                'id',   
+                                'name',
+                                'parent_id',
+                                'created_at',
```
Comment (at-thinhnguyen) [inline tests/Feature/Api/listCategoriesTest.php]: sao chổ này có created_at, updated_at nhỉ, docs ko có sao nó vẫn pass @@

File: tests/Feature/Api/listCategoriesTest.php [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    const NUMBER_RECORD_CREATE = 5;
+ 
+    /**
+     * Receive status code 200 when get list categories success.
+     *
+     * @return void
+     */
+    public function testStatusCodeSuccess()
+    {
+        factory(Category::class, 'parent', 1)->create();
```
Comment (at-thinhnguyen) [inline tests/Feature/Api/listCategoriesTest.php]: nên tạo nhiều data để test thêm children

**Pull request merged**